llvm/lib/CodeGen/CFIInstrInserter.cpp

```diff
@@ -267,11 +278,11 @@ void CFIInstrInserter::calculateOutgoingCFAInfo(MBBCFAInfo &MBBInfo) {
         break;
       }
       if (CSRReg || CSROffset) {
+        CSRSavedLocation Loc(CSRReg, CSROffset);
         auto It = CSRLocMap.find(CFI.getRegister());
```

Comment:
We should try the `insert` instead of `find` here. If the `insert` fails, do the check to make sure the value we tried to insert matches what was already there.

This can be a follow up.